### PR TITLE
[Lint] Fix no typedef errors

### PR DESCRIPTION
**Pull request by Shobhit1**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000 - N/A
- [ ] Added relevant unit test for your changes. (`npm run test`)- N/A
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`- N/A
- [x] Ran precheckin (`npm run precheckin`)
- [ ] Added screenshots/GIFs for UI changes.- N/A

#### Description of changes 

Lint Errros after this PR: **342 ▶️  273**

### CODE REVIEW

File: src/common/components/selector-input-list.tsx [hunk 1]
```diff
@@ -146,14 +146,14 @@ export class SelectorInputList extends React.Component<SelectorInputListProps, S
     }
 
     @autobind
-    private getDeleteSelectorHandler(item: string[]) {
+    private getDeleteSelectorHandler(item: string[]): (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void {
```
Comment (smoralesd) [inline src/common/components/selector-input-list.tsx]: do we need the `MouseEvent` for the type?

File: src/injected/analyzer-controller.ts [hunk 2]
```diff
@@ -95,7 +95,7 @@ export class AnalyzerController {
         });
     }
 
-    private getAnalyzerByIdentifier(key: string) {
+    private getAnalyzerByIdentifier(key: string): IAnalyzer<string> {
```
Comment (smoralesd) [inline src/injected/analyzer-controller.ts]: This is a little tricky by I think `<string>` is not the correct type for the `IAnalyzer` return type here.

Comment (Shobhit1) [reply src/injected/analyzer-controller.ts]: We are returning `IAnalyzer` using `key: string`. That was my thinking for this

File: src/common/components/selector-input-list.tsx (hunk 1 above)
Comment (Shobhit1) [reply src/common/components/selector-input-list.tsx]: Updated.

File: src/injected/analyzer-controller.ts (hunk 2 above)
Comment (Shobhit1) [reply src/injected/analyzer-controller.ts]: Updated.

**Pull request merged**